src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommandBuilder.cs

```diff
@@ -0,0 +1,301 @@
+            get
+            {
+                return base.QuoteSuffix;
+            }
+            set
+            {
+                if (("]" != value) && ("\"" != value))
+                {
+                    throw ADP.DoubleValuedProperty(nameof(QuoteSuffix), "]", "\"");
+                }
+                base.QuoteSuffix = value;
+            }
+        }
+
+        public override string SchemaSeparator
+        {
+            get
+            {
+                return ".";
+            }
+            set
+            {
+                if ("." != value)
+                {
+                    throw ADP.SingleValuedProperty(nameof(SchemaSeparator), ".");
+                }
+            }
+        }
+
+        private void SqlRowUpdatingHandler(object sender, SqlRowUpdatingEventArgs ruevent)
+        {
+            base.RowUpdatingHandler(ruevent);
+        }
+
+        new public SqlCommand GetInsertCommand()
+        {
+            return (SqlCommand)base.GetInsertCommand();
+        }
+
+        new public SqlCommand GetInsertCommand(bool useColumnsForParameterNames)
+        {
+            return (SqlCommand)base.GetInsertCommand(useColumnsForParameterNames);
+        }
+
+        new public SqlCommand GetUpdateCommand()
+        {
+            return (SqlCommand)base.GetUpdateCommand();
+        }
+
+        new public SqlCommand GetUpdateCommand(bool useColumnsForParameterNames)
+        {
+            return (SqlCommand)base.GetUpdateCommand(useColumnsForParameterNames);
+        }
+
+        new public SqlCommand GetDeleteCommand()
+        {
+            return (SqlCommand)base.GetDeleteCommand();
+        }
+
+        new public SqlCommand GetDeleteCommand(bool useColumnsForParameterNames)
+        {
+            return (SqlCommand)base.GetDeleteCommand(useColumnsForParameterNames);
+        }
+
+        protected override void ApplyParameterInfo(DbParameter parameter, DataRow datarow, StatementType statementType, bool whereClause)
+        {
+            SqlParameter p = (SqlParameter)parameter;
+            object valueType = datarow[SchemaTableColumn.ProviderType];
+            p.SqlDbType = (SqlDbType)valueType;
+            p.Offset = 0;
+
+            object bvalue = datarow[SchemaTableColumn.NumericPrecision];
+            if (DBNull.Value != bvalue)
+            {
+                byte bval = (byte)(short)bvalue;
+                p.PrecisionInternal = ((0xff != bval) ? bval : (byte)0);
+            }
+
+            bvalue = datarow[SchemaTableColumn.NumericScale];
+            if (DBNull.Value != bvalue)
+            {
+                byte bval = (byte)(short)bvalue;
+                p.ScaleInternal = ((0xff != bval) ? bval : (byte)0);
+            }
+        }
+
+        protected override string GetParameterName(int parameterOrdinal)
+        {
+            return "@p" + parameterOrdinal.ToString(CultureInfo.InvariantCulture);
+        }
+
+        protected override string GetParameterName(string parameterName)
```

Comment:
Nit: Can be `protected override string GetParameterName(string parameterName) => "@" + parameterName;`